dvc/output.py

```diff
@@ -396,13 +398,19 @@ def __init__(  # noqa: C901
             name=self.hash_name,
             value=getattr(self.meta, self.hash_name, None),
         )
+        self._compute_meta_hash_info_from_files()
+
+    def _compute_meta_hash_info_from_files(self) -> None:
         if self.files:
             tree = Tree.from_list(self.files, hash_name=self.hash_name)
             tree.digest()
             self.hash_info = tree.hash_info
             self.meta.isdir = True
             self.meta.nfiles = len(self.files)
             self.meta.size = sum(file.get("size") for file in self.files)
+            self.meta.remote = first(
+                first(file.get("cloud", {})) for file in self.files
+            )
```

Comment:
All of this serialization/deserialization logic should be outside. Will try to look into it after this PR. It's not very important, but it feels like we are struggling with this, as we have to maintain `.dvc` and `dvc.lock` files.